test/integration/analytical_rosenbrock.cpp

```diff
@@ -104,243 +104,45 @@ using LinearSolverTest = micm::LinearSolver<SparseMatrixTest, micm::LuDecomposit
-    { 0.1043985088527474E+01, 0.2373442027531524E+02, 0.1037691843544522E+01 },
-    { 0.1100849071667922E+01, 0.1091533805469020E+02, 0.1085831969810860E+01 },
-    { 0.1249102130020572E+01, 0.5013945178605446E+01, 0.1208326626237875E+01 },
-    { 0.1779724751937019E+01, 0.2281852385542403E+01, 0.1613754023671725E+01 },
-    { 0.1000889326903503E+01, 0.1125438585746596E+04, 0.1641049483777168E+05 },
-    { 0.1000814870318523E+01, 0.1228178521549889E+04, 0.1320554942846513E+03 },
-  };
-
-  auto state = solver.rates_.GetState();
 
-  state.variables_[0] = model_concentrations[0];
+  auto rosenbrock_solver = OregonatorTest::template CreateSolver<RosenbrockTest<OregonatorTest>, LinearSolverTest>(params, 1);
+  auto backward_euler_params = micm::BackwardEulerSolverParameters();
+  double abolute_tolerance = 1e-16;
+  backward_euler_params.absolute_tolerance_ = { abolute_tolerance, abolute_tolerance, abolute_tolerance };
+  backward_euler_params.relative_tolerance_ = 1e-10;
+  auto backward_euler_solver = OregonatorTest::template CreateSolver<BackwardEulerTest<OregonatorTest>, LinearSolverTest>(backward_euler_params, 1);
 
-  std::vector<double> times;
-  times.push_back(0);
-  for (size_t i_time = 0; i_time < N; ++i_time)
-  {
-    double solve_time = time_step + i_time * time_step;
-    times.push_back(solve_time);
-    // Model results
-    double actual_solve = 0;
-    while (actual_solve < time_step)
-    {
-      auto result = solver.Solve(time_step - actual_solve, state);
-      state.variables_[0] = state.variables_.AsVector();
-      actual_solve += result.final_time_;
-    }
-    model_concentrations[i_time + 1] = state.variables_[0];
-  }
-
-  std::vector<std::string> header = { "time", "A", "B", "C" };
-  writeCSV("model_concentrations.csv", header, model_concentrations, times);
-  std::vector<double> an_times;
-  an_times.push_back(0);
-  for (int i = 1; i <= 12; ++i)
-  {
-    an_times.push_back(30 * i);
-  }
-  writeCSV("analytical_concentrations.csv", header, analytical_concentrations, an_times);
-
-  auto map = state.variable_map_;
-
-  size_t _a = map.at("A");
-  size_t _b = map.at("B");
-  size_t _c = map.at("C");
-
-  double tol = 1e-3;
-  for (size_t i = 0; i < model_concentrations.size(); ++i)
-  {
-    double rel_diff = relative_difference(model_concentrations[i][_a], analytical_concentrations[i][0]);
-    EXPECT_TRUE(rel_diff < tol) << "Arrays differ at index (" << i << ", " << 0 << ")";
-    rel_diff = relative_difference(model_concentrations[i][_b], analytical_concentrations[i][1]);
-    EXPECT_TRUE(rel_diff < tol) << "Arrays differ at index (" << i << ", " << 1 << ")";
-    rel_diff = relative_difference(model_concentrations[i][_c], analytical_concentrations[i][2]);
-    EXPECT_TRUE(rel_diff < tol) << "Arrays differ at index (" << i << ", " << 2 << ")";
-  }
+  test_analytical_oregonator(rosenbrock_solver, 1e-3);
+  // test_analytical_oregonator(backward_euler_solver, 1e-1);
```

Comment:
Does the backward euler solver not work for the Oregonator test?

Reply:
It does not. It's not even close to working. The very first time step we already have almost a 100% relative difference

Reply:
For reference, these are the equations that compose the Oregonator

<img width="806" alt="Screenshot 2024-06-26 at 12 55 25 PM" src="https://github.com/NCAR/micm/assets/13836216/4f522e32-4a5b-4dc6-aae1-eebd6e17ec9a">

Image from 
Hairer, E., Wanner, G., 1996. Solving Ordinary Differential Equations II: Stiff and Differential-Algebraic Problems, 2nd edition. ed. Springer, Berlin ; New York.

Reply:
Thanks for the additional information. Not sure why it does not work for the BE solver but this may highlight the importance of using high-order numerical solver for the chemistry problem.

Reply:
Probably just remove this test for BE solver since it is not working?

Reply:
removed